docs(partials): add info about using a parital with custom plugin (#8608)
Kong/docs.konghq.com: app/_src/gateway/kong-enterprise/partials.md

- **`redis-ce`**: Stores common Redis configurations for plugins available in {{site.ce_product_name}}.
- **`redis-ee`**: Stores common Redis configurations for plugins available only in {{site.ee_product_name}}.

> An easy way to determine the required Redis configuration is by examining the fields in use. `redis-ce` has a shorter and simpler configuration, whereas `redis-ee` provides options for configuring Redis Sentinel or Redis Cluster for connection.

Reviewer: ```suggestion
An easy way to determine the required Redis configuration is by examining the fields in use. `redis-ce` has a shorter and simpler configuration, whereas `redis-ee` provides options for configuring Redis Sentinel or Redis Cluster for connections.
```

- **`redis-ce`**: Stores common Redis configurations for plugins available in {{site.ce_product_name}}.
- **`redis-ee`**: Stores common Redis configurations for plugins available only in {{site.ee_product_name}}.

> {:.note}
> > An easy way to determine the required Redis configuration is by examining the fields in use. Redis CE has a shorter and simpler configuration, whereas Redis EE provides options for configuring Redis Sentinel or Redis Cluster for connection.

Reviewer: ```suggestion
An easy way to determine the required Redis configuration is by examining the fields in use. `redis-ce` has a shorter and simpler configuration, whereas `redis-ee` provides options for configuring Redis Sentinel or Redis Cluster for connection.
```
This doesn't need to be a note, it can just be a regular paragraph.

curl -i -X DELETE http://localhost:8001/partials/{PARTIAL-ID}
```

> ## Custom Plugins

Reviewer: ```suggestion
## Custom plugins
```

> Users can leverage the Partials feature in their custom plugins by adjusting the plugin schema.

Reviewer: ```suggestion
You can leverage the Partials feature in your custom plugins by adjusting the plugin schema.
```

To make custom plugins compatible with Partials, add the `supported_partials` key to the schema and specify
the appropriate Partial type. If the Redis configuration in use is of the {{site.ce_product_name}} type,
use `redis-ce`; otherwise, use `redis-ee`.

> Below is an example schema for a custom plugin utilizing a Partial:

Reviewer: ```suggestion
Here is an example schema for a custom plugin using a Partial:
```

```lua
{
  name = "custom-plugin-with-redis",
  supported_partials = {
    ["redis-ee"] = { "config.redis" },
  },
  fields = {
    {
      config = {
        type = "record",
        fields = {
          { some_other_config_key = { type = "string", required = true }},
          { redis = redis.config_schema }
        },
      },
    },
  },
}
```

> {:.important} > **Using DAO in custom plugins**

Reviewer: `important` is right here, it'll be a yellow note. I'm good with that.

> {:.important} > **Using DAO in custom plugins**
> > Be aware that when using a Partial, the configuration belonging to the Partial is no longer stored alongside
> > the plugin. If your code relies on Kong's dao and expects entities to contain Redis information,
> > this data will not be retrieved when using `kong.db.plugins:select(plugin_id)`.
> > Such a call will only fetch data stored in the plugin itself.
> >
> > To include the Partial's data within the plugin configuration, pass a special option parameter,
> > such as: `kong.db.plugins:select(plugin_id, { expand_partials = true })`.

Reviewer: ```suggestion
{:.important} 
> **Using DAO in custom plugins**
> <br>
> Be aware that when using a Partial, the configuration belonging to the Partial is no longer stored alongside
> the plugin. If your code relies on {{site.base_gateway}}'s DAO and expects entities to contain Redis information,
> this data won't be retrieved when using `kong.db.plugins:select(plugin_id)`.
> Such a call will only fetch data stored in the plugin itself.
>
> To include the Partial's data within the plugin configuration, you must pass a special option parameter,
> such as: `kong.db.plugins:select(plugin_id, { expand_partials = true })`.
```
